dpnp/dpnp_iface_statistics.py

```diff
@@ -634,70 +634,204 @@ def ptp(
+        will have the default floating point data type for the device
+        where input array `a` is allocated.
+
     Limitations
     -----------
-    Input array is supported as :obj:`dpnp.ndarray`.
-    Parameter `axis` is supported only with default value ``None``.
-    Parameter `dtype` is supported only with default value ``None``.
-    Parameter `out` is supported only with default value ``None``.
-    Parameter `keepdims` is supported only with default value ``False``.
-    Otherwise the function will be executed sequentially on CPU.
+    Parameters `where` is only supported with its default value.
+    Otherwise ``NotImplementedError`` exception will be raised.
+    Input array data types are limited by real valued data types.
+
+    See Also
+    --------
+    :obj:`dpnp.var` : Compute the variance along the specified axis.
+    :obj:`dpnp.nanmean` : Compute the arithmetic mean along the specified axis,
+                          ignoring NaNs.
+    :obj:`dpnp.nanstd` : Compute the standard deviation along
+                         the specified axis, while ignoring NaNs.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> a = np.array([[1, np.nan], [3, 4]])
+    >>> np.nanvar(a)
+    array(1.5555555555555554)
+    >>> np.nanvar(a, axis=0)
+    array([1.,  0.])
+    >>> np.nanvar(a, axis=1)
+    array([0.,  0.25])  # may vary
+
     """
 
-    x1_desc = dpnp.get_dpnp_descriptor(x1, copy_when_nondefault_queue=False)
-    if x1_desc:
-        if x1.size == 0:
-            pass
-        elif axis is not None:
-            pass
-        elif dtype is not None:
-            pass
-        elif out is not None:
-            pass
-        elif keepdims:
-            pass
-        else:
-            result_obj = dpnp_nanvar(x1_desc, ddof).get_pyobj()
-            result = dpnp.convert_single_elem_array_to_scalar(result_obj)
+    if where is not True:
+        raise NotImplementedError(
+            "where keyword argument is only supported with its default value."
+        )
 
-            return result
+    arr, mask = dpnp._replace_nan(a, 0)
+    if mask is None:
+        return dpnp.var(
+            arr,
+            axis=axis,
+            dtype=dtype,
+            out=out,
+            ddof=ddof,
+            keepdims=keepdims,
+            where=where,
+        )
 
-    return call_origin(
-        numpy.nanvar,
-        x1,
-        axis=axis,
-        dtype=dtype,
-        out=out,
-        ddof=ddof,
-        keepdims=keepdims,
+    if issubclass(arr.dtype.type, dpnp.complexfloating):
+        raise ValueError("`nanvar` does not support complex types")
+    if dtype is not None:
+        dtype = dpnp.dtype(dtype)
+        if not issubclass(dtype.type, dpnp.floating):
+            raise TypeError("If input is floating, then dtype must be floating")
+    if out is not None and not issubclass(out.dtype.type, dpnp.inexact):
+        raise TypeError("If input is inexact, then out must be inexact")
+
+    # Compute mean
```

Comment:
By that implementation, the resulting dtype might be different with `dpnp.var` call in some cases, please check the example:
```python
a = dpnp.arange(10, dtype=numpy.float32, device='cpu')
dpnp.var(a).dtype
# Out: dtype('float32')

dpnp.nanvar(a).dtype
# Out: dtype('float64')

a[::2] = dpnp.nan
dpnp.nanvar(a).dtype
# Out: dtype('float64')
```
I wonder if it is expected or not.